p2panda-core/src/prune.rs

```diff
@@ -1,9 +1,43 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
+use std::ops::Deref;
+
 use serde::{Deserialize, Serialize};
 
 use crate::{validate_backlink, Header, OperationError};
 
+#[derive(Clone, Debug, Default, Hash, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct PruneFlag(bool);
+
+impl PruneFlag {
+    pub fn new(flag: bool) -> Self {
+        Self(flag)
+    }
+
+    pub fn is_set(&self) -> bool {
+        self.0
+    }
+
+    pub fn is_not_set(&self) -> bool {
+        !self.0
+    }
```

Comment:
This method is a bit funny, but helps to have a nice `serde` usage, not sure how to do it better

Reply:
I left a suggestion in the test code.